SalesforceSDK/SmartStore/SmartStore.cs

```diff
@@ -0,0 +1,1132 @@
+
+        // Columns of the soup index map table
+        public static readonly string SoupNameCol = "soupName";
+        public static readonly string PathCol = "path";
+        public static readonly string ColumnNameCol = "columnName";
+        public static readonly string ColumnTypeCol = "columnType";
+
+        // Columns of a soup table
+        public static readonly string IdCol = "id";
+        public static readonly string CreatedCol = "created";
+        public static readonly string LastModifiedCol = "lastModified";
+        public static readonly string SoupCol = "soup";
+
+        // Columns of Lon operations status table
+        public static readonly string TypeCol = "type";
+        public static readonly string DetailsCol = "details";
+        public static readonly string StatusCol = "status";
+
+        // JSON fields added to soup element on insert/update
+        public static readonly string SoupEntryId = "_soupEntryId";
+        public static readonly string SoupLastModifiedDate = "_soupLastModifiedDate";
+
+        // Predicates
+        public static readonly string SoupNamePredicate = SoupNameCol + " = ?";
+        public static readonly string PathPredicate = PathCol + " = ?";
+        public static readonly string IdPredicate = IdCol + " = ?";
+
+        private static readonly object smartlock = new object();
+        private static readonly Regex SmartSqlRegex = new Regex("\\{([^}]+)\\}", RegexOptions.IgnoreCase);
+        private static readonly DateTime Jan1st1970 = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        // Backing database
+        private string _databasePath;
+
+        public string DatabasePath
+        {
+            get
+            {
+                return _databasePath;
+            }
+        }
+
+        public DBHelper Database
+        {
+            get { return DBHelper.GetInstance(DatabasePath); }
+        }
+
+        public static long CurrentTimeMillis
+        {
+            get { return (long) DateTime.UtcNow.Ticks; }
+        }
+
+        private SmartStore()
+        {
+            _databasePath = GenerateDatabasePath(AccountManager.GetAccount());
+        }
+
+        private SmartStore(Account account)
+        {
+            _databasePath = GenerateDatabasePath(account);
+        }
+
+        public static SmartStore GetSmartStore()
+        {
+            var store = new SmartStore();
+            store.CreateMetaTables();
+            return store;
+        }
+
+        public static SmartStore GetSmartStore(Account account)
+        {
+            var store = new SmartStore(account);
+            store.CreateMetaTables();
+            return store;
+        }
+
+        public static SmartStore GetGlobalSmartStore()
+        {
+            var store = new SmartStore(null); // generate a "global" smartstore
+            store.CreateMetaTables();
+            return store;
+        }
+
+        public static string GenerateDatabasePath(Account account)
```

Comment:
Assuming you just brought this code over from somewhere else (or is this new code?)... but why have the "account" parameter, doesn't look like it's used.